src/invidious/helpers/signatures.cr

```diff
@@ -10,7 +10,16 @@ struct Invidious::DecryptFunction
 
   def check_update
     now = Time.monotonic
-    if (now - @last_update) > 60.seconds
+
+    # If we have updated in the last 5 minutes, do nothing
+    return if (now - @last_update) > 5.minutes
+
+    # Get the time when the player was updated, in the event where
+    # multiple invidious processes are run in parallel.
+    player_ts = Invidious::SigHelper::Client.get_player_timestamp
+    player_time = Time.unix(player_ts || 0)
+
+    if (now - player_time) > 5.minutes
```

Comment:
at least according to https://github.com/iv-org/inv_sig_helper#player_update_timestamp-0x05, the player update timestamp is already seconds since the last update, so this should be something like

```
player_time = (player_ts || 42).minutes

if player_time > 5.minutes
```

Reply:
Damn, I should learn to read :x I was expecting a unix timestamp here....